apis/v1alpha1/gatewayclass_types.go

```diff
@@ -85,10 +85,21 @@ type GatewayClassSpec struct {
 type GatewayClassConditionType string
 
 const (
-	// GatewayClassConditionStatusInvalidParameters indicates the
-	// validity of the Parameters set for a given controller. This
-	// will initially start off as "Unknown".
-	GatewayClassConditionStatusInvalidParameters GatewayClassConditionType = "InvalidParameters"
+	// GatewayClassConditionStatusAccepted indicates that the GatewayClass
+	// has been accepted by the controller set in the `spec.controller`
+	// field.
+	//
+	// It defaults to Unknown, and MUST be set by a controller when it sees
+	// a GatewayClass using its controller string.
+	// It must be set to `status: true` if the controller is using the class,
+	// and `status:false` if the controller is not using the class for some reason.
+	// In either case, the controller SHOULD set a Message and Reason as an explanation.
+	// GatewayClassNotAcceptedInvalidParameters is provided as an example Reason.
```

Comment:
Do we need a message and reason if status == true?

Reply:
I wasn't sure, that makes sense to add.

Reply:
I think it's also fine + potentially simpler to say "If the status is set to false, the controller SHOULD set a Message and Reason as an explanation". I'm not sure we need any kind of message reason for the true case.